### ISSUE TITLE
Add autolinking compatibility

### ISSUE THREAD

**Issue by sattaman** [OneSignal/react-native-onesignal#946]
**Description:**

Following the instructions I get the following error, if I don't expicitly link your module does not get found so I am presumming this isn't yet compatible with autolinking? 

> error React Native CLI uses autolinking for native dependencies, but the following modules are linked manually: 
> 
>   - react-native-onesignal (to unlink run: "react-native unlink react-native-onesignal")
> 
> This is likely happening when upgrading React Native from below 0.60 to 0.60 or above. Going forward, you can unlink this dependency via "react-native unlink <dependency>" and it will be included in your app automatically. If a library isn't compatible with autolinking, disregard this message and notify the library maintainers.
> Read more about autolinking: https://github.com/react-native-community/cli/blob/master/docs/autolinking.md

**Environment**
react-native 0.61.4
react-native-onesignal: 3.6.3

**Comment by ricardojr** [comment]
Same here
react-native: 0.61.5
react-native-onesignal: 3.6.4

**Comment by rborn** [comment]
@ricardojr I didn't do "react-native link" when I installed the sdk, just followed all the other steps except link and it worked.

**Comment by rgomezp** [comment]
My understanding is that the only requirement to be able to "auto-link" is including a Podfile and gradle file. React-native should automatically do everything else. Since this has worked for most people in the past, I suspect this issue is because of another problem.

**Comment by adambeck7** [comment]
My autolinking isn't working either. @rgomezp I'm trying to link manually but I get this error when I try to manually link in mainApplication.java `package com.geektime.rnonesignalandroid does not exist` Is the import no longer `import com.geektime.rnonesignalandroid.ReactNativeOneSignalPackage;`?

**Comment by jkasten2** [comment]
@adambeck7 Do you see this error when running `react-native link react-native-onesignal` or when your building? Can you share your full output too?

If linking worked you should see `implementation project(':react-native-onesignal')` in your `app/build.gradle`
https://github.com/OneSignal/react-native-onesignal/blob/3.6.5/examples/RNOneSignal/android/app/build.gradle#L191

**Comment by adambeck7** [comment]
@jkasten2  It was when building, not linking.  It would say it linked and unlinked successfully but we couldn't get it to work on 3.6.4 with React-Native 0.61.2. When we switched to 3.6.5 it started working. 

Only issue now is that we removed our OneSignal.configure() and now we can't get our Ids. OneSignal.getPermissionSubscriptionState() returns undefined like issue #296 . Everything else is working fine.
